src/components/foodDescription/FoodDescription.js

```diff
@@ -1,44 +1,159 @@
             foodListSectionRefs.current.push(ele);
         }
     }
 
     function handleCategoryScroll(category) {
-        const section = _find(foodListSectionRefs.current, { id: _toLower(category) });
+        const section = _find(foodListSectionRefs.current, {
+            id: _toLower(category),
+        });
         section.scrollIntoView({ behavior: "smooth" });
     }
 
+    function handleAddToCart(event) {
+        const {
+            target: {
+                dataset: { id },
+            },
+        } = event;
+        const isIdPresentInCart = _has(cart, id);
+        if (!isIdPresentInCart) {
+            setCart({ ...cart, [id]: 1 });
+        }
+    }
+
+    function handleIncrementClick(event) {
+        const {
+            target: {
+                parentNode: {
+                    dataset: { id },
+                },
+            },
+        } = event;
+        const newQuantitiy = cart[id] + 1;
+        setCart({ ...cart, [id]: newQuantitiy });
+    }
+
+    function handleDecrementClick(event) {
+        const {
+            target: {
+                parentNode: {
+                    dataset: { id },
+                },
+            },
+        } = event;
+        const newQuantitiy = cart[id] - 1;
+        if (newQuantitiy === 0) {
+            setCart(_omit(cart, id));
+        } else {
+            setCart({ ...cart, [id]: newQuantitiy });
+        }
+    }
+
+    function handleCheckoutClick() {
+        const items = getCartItems(cart, foodList);
+        const amount = getTotalFromCartItems(items);
+        const cartItemsData = { amount, items };
+        saveToLocalStorage(CART_ITEMS, cartItemsData);
+        clearCart();
+    }
+
+    function clearCart() {
+        setCart({});
+    }
+
+    function renderCart() {
+        const isCartEmpty = _isEmpty(cart);
+        if (isCartEmpty) {
+            return (
+                <>
+                    <h2>Cart Empty</h2>
+                    <Icon
+                        className="empty-cart-img"
+                        src={emptyCartImage}
+                        alt="empty-icon"
+                    />
+                    <p>{EMPTY_CART_DESCRIPTION}</p>
+                </>
+            );
+        }
+        return (
+            <Cart
+                cart={cart}
+                foodList={foodList}
+                handleIncrementClick={handleIncrementClick}
+                handleDecrementClick={handleDecrementClick}
+                handleCheckoutClick={handleCheckoutClick}
+            />
+        );
+    }
+
     return (
         <div className="flex food-description">
             <Sidebar
-                categories={keys}
+                categories={categories}
                 handleCategoryScroll={handleCategoryScroll}
             />
-            <FoodList foodList={foodList} setRef={setRef} />
+            <FoodList
+                foodList={foodList}
+                cart={cart}
+                handleAddToCart={handleAddToCart}
```

Comment:
function declaration should start with handle and props should start with on.. for event handlers